Client/Shared/MainLayout.razor

```diff
@@ -5,18 +5,16 @@
 @implements IDisposable
 
 <div class="sidebar">
-    <NavMenuComponent EventCallback="HandleThemeChange" />
+    <NavMenuComponent EventCallback="HandleThemeChange"/>
 </div>
 <div class="page @(isWhiteTheme ? "light-mode" : "dark-mode")">
-    <div class="topbar @(isWhiteTheme ? "" : "topbar-dark")">
-        <CascadingValue Value="isWhiteTheme">
-            <TopBarComponent />
-        </CascadingValue>
+    <div class="topbar @(isWhiteTheme ? "light-mode" : "dark-mode")">
```

Comment:
old style for dark-mode should only be applied for (div class="page")
please remove: @(isWhiteTheme ? "light-mode" : "dark-mode") from (div class="topbar")

Reply:
Why you want two different mechanism for same functionality?

Reply:
I mean you have to apply old mechanism to global div (page).
If you don't the new mechanism will not work.

Reply:
So it should look like that:

div class="page @(isWhiteTheme ? "light-mode" : "dark-mode")"
    div class="topbar"

Reply:
As i saw in css, you are not even using the mechanism you left in code so

Reply:
.css has a new version
.razor has an old one

.light-mode .topbar {
    background: white;
    box-shadow: -3px -13px 15px 13px #ccc;
}

.dark-mode .topbar {
    background: #3F4D54;
    box-shadow: -3px -13px 15px 13px #333F44;
}